packages/tools/src/tools/annotation/splines/CubicSpline.ts

```diff
@@ -0,0 +1,217 @@
+import { Types } from '@cornerstonejs/core';
+import { Spline } from './Spline';
+import * as math from '../../../utilities/math';
+import type { SplineCurveSegment } from './types/SplineCurveSegment';
+import type { SplineLineSegment } from './types/SplineLineSegment';
+
+abstract class CubicSpline extends Spline {
+  protected getSplineCurves(): SplineCurveSegment[] {
+    const numCurveSegments = this.getNumCurveSegments();
+    const curveSegments: SplineCurveSegment[] = new Array(numCurveSegments);
+
+    if (numCurveSegments <= 0) {
+      return [];
+    }
+
+    const transformMatrix = this.getTransformMatrix();
+    let lengthStart = 0;
+
+    for (let i = 0; i < numCurveSegments; i++) {
+      const { p0, p1, p2, p3 } = this._getCurveSegmentPoints(i);
+      const lineSegments = this._getLineSegments(i, transformMatrix);
+      let curveSegmentLength = 0;
+      let minX = Infinity;
+      let minY = Infinity;
+      let maxX = -Infinity;
+      let maxY = -Infinity;
+
+      lineSegments.forEach((lineSegment) => {
+        const { aabb: lineSegAABB } = lineSegment;
+
+        minX = minX <= lineSegAABB.minX ? minX : lineSegAABB.minX;
+        minY = minY <= lineSegAABB.minY ? minY : lineSegAABB.minY;
+        maxX = maxX >= lineSegAABB.maxX ? maxX : lineSegAABB.maxX;
+        maxY = maxY >= lineSegAABB.maxY ? maxY : lineSegAABB.maxY;
+        curveSegmentLength += lineSegment.length;
+      });
```

Comment:
How about this 

```js
let minX = Infinity;
let minY = Infinity;
let maxX = -Infinity;
let maxY = -Infinity;

lineSegments.forEach(({ aabb: lineSegAABB }) => {
  minX = Math.min(minX, lineSegAABB.minX);
  minY = Math.min(minY, lineSegAABB.minY);
  maxX = Math.max(maxX, lineSegAABB.maxX);
  maxY = Math.max(maxY, lineSegAABB.maxY);
});
}
```

Reply:
That works as well but if you have noticed I also don't use to use `.map(...)`, `.forEach(...)`  when I'm working on loops that can run hundreds of times to reduce function calls because switching between contexts when invoking subroutines requires more clock cycles in assembly. I started working on that way after investigating why a cine player was not able to play at 60fps and the annotations was one of the problems.